dvc/output/local.py

```diff
@@ -18,6 +18,12 @@ class OutputLOCAL(OutputBase):
     REMOTE = RemoteLOCAL
     sep = os.sep
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.repo and path_isin(self.def_path, self.repo.root_dir):
+            self.def_path = relpath(self.def_path, self.stage.wdir)
```

Comment:
@casperdcl One more thing. Just noticed that moving this logic above `super()` would actually be more robust, as we would convert `path` before `OutputBase.__init__` tries to do anything with it (e.g. imagine it starting to use `is_in_repo` at some point, before we've converted the path). So moving it would make the code more robust.

Reply:
Plus you won't need to test OutputBASE as you do below :smiley:

Reply:
right moved to pre-`super()` now. Not sure about removing the test though since it tests logic regardless of the fact that the current (code) implementation guarantees the test passes. I mean, all tests are guaranteed to pass if the code is working as expected, right?